src/main/java/in/org/projecteka/hdaf/link/link/ClientError.java

```diff
@@ -0,0 +1,17 @@
+package in.org.projecteka.hdaf.link.link;
+
+import in.org.projecteka.hdaf.link.link.model.ErrorRepresentation;
+import lombok.Getter;
+import org.springframework.http.HttpStatus;
+
+@Getter
+public class ClientError extends Throwable {
+
+    private HttpStatus httpStatus;
```

Comment:
I Prefer to not keep the HttpStatus in the service level classes. Not a big deal though.